examples/js/utils/BufferGeometryUtils.js

```diff
@@ -620,23 +612,7 @@ THREE.BufferGeometryUtils = {
 
 		}
 
-		// Generate an index buffer typed array
-		var cons = Uint8Array;
-		if ( newIndices.length >= Math.pow( 2, 8 ) ) cons = Uint16Array;
-		if ( newIndices.length >= Math.pow( 2, 16 ) ) cons = Uint32Array;
-
-		var newIndexBuffer = new cons( newIndices );
-		var newIndices = null;
-		if ( indices === null ) {
-
-			newIndices = new THREE.BufferAttribute( newIndexBuffer, 1 );
-
-		} else {
-
-			newIndices = geometry.getIndex().clone();
-			newIndices.setArray( newIndexBuffer );
-
-		}
```

Comment:
I think this entire code section is not necessary. `BufferGeometry.setIndex()` already ensures to pick the correct buffer attribute type (`Uint32BufferAttribute` vs. `Uint16BufferAttribute`).

Reply:
> FWIW setIndex never uses Uint8Array, only Uint16 or Uint32. We discourage Uint8 indices in glTF because it isn't supported by DirectX, although I don't know if that affects WebGL in any way.

I'm sure I've used Uint8 for indices to save CPU memory before without issue on windows before, which is where I think this is most relevant. I'm not super worried about this, though, because I think merging meshes would often result in a mesh with more than 256 vertices, anyway.

It would be nice to know if Uint8 indices are not well supported, though. If they are then it might be a nice addition for `setIndex` to support them.

Reply:
> It would be nice to know if Uint8 indices are not well supported, though. If they are then it might be a nice addition for setIndex to support them.

Good idea.

Reply:
I would be curious to know, but it's a _very_ small optimization since we're only dealing with <=256 vertices at that point. 😅

Reply:
Heh I suppose that is true -- it seemed like a natural addition but maybe it's too insignificant.